### PR TITLE
Support available_time_ranges in dashboard config

### PR DESCRIPTION
**Pull request by AdityaHegde**
## Checklist
- [ ] Manual verification
- [ ] Unit test coverage
- [ ] E2E test coverage
- [ ] Needs manual QA?

### CODE REVIEW

File: runtime/compilers/rillv1/parser_test.go [hunk 1]
```diff
@@ -94,6 +94,13 @@ measures:
     expression: count(*)
 first_day_of_week: 7
 first_month_of_year: 3
+available_time_ranges:
+  - P2W
+  - range: P4W
+  - range: P2M
+    comparison_ranges:
+      - P1M
+      - P1Y
```
Comment (begelundmuller) [inline runtime/compilers/rillv1/parser_test.go:103]: Are these comparison ranges or comparison offsets?

File: runtime/compilers/rillv1/parse_metrics_view.go [hunk 2]
```diff
@@ -6,29 +6,28 @@ import (
 	"fmt"
 	"strings"
 	"time"
+	// Load IANA time zone data
+	_ "time/tzdata"
```
Comment (begelundmuller) [inline runtime/compilers/rillv1/parse_metrics_view.go:10]: Formatting err

File: proto/rill/runtime/v1/resources.proto [hunk 3]
```diff
@@ -180,6 +184,8 @@ message MetricsViewSpec {
   uint32 first_day_of_week = 12;
   // Month number to use as the base for time aggregations by year. Defaults to 1 (January).
   uint32 first_month_of_year = 13;
+  // List of available time ranges with comparison ranges that would replace the default list.
+  repeated AvailableTimeRange available_time_ranges = 14;
```
Comment (begelundmuller) [inline proto/rill/runtime/v1/resources.proto:188]: Need to merge main; conflicts with `default_comparison_mode`

**Review by begelundmuller — CHANGES_REQUESTED** [review]

File: runtime/compilers/rillv1/parser_test.go (hunk 1 above)
Comment (AdityaHegde) [reply runtime/compilers/rillv1/parser_test.go:103]: These are just for testing. Actual values are pending a decision on the format.

File: runtime/compilers/rillv1/parse_metrics_view.go (hunk 2 above)
Comment (AdityaHegde) [reply runtime/compilers/rillv1/parse_metrics_view.go:10]: Ya will fix these just before merging.

File: runtime/compilers/rillv1/parser_test.go (hunk 1 above)
Comment (begelundmuller) [reply runtime/compilers/rillv1/parser_test.go:103]: I meant that if they are offsets, then I think we should make the key `comparison_offsets` instead :) This sounds like you're comparing e.g. 2 months of revenue vs. 1 month or 1 year of revenue – I'm missing context, but I don't think that's what we're doing?

**Comment by AdityaHegde** [comment]
Closing in favour of #3418 for backend changes. UI changes will come in a later PR after #3418 and #3417 are done.

**Pull request closed without merging**